src/frontend/static/frontend/src/components/pipeline/all-experiments-view/ClinicalSourcePopup.tsx

```diff
@@ -564,7 +577,7 @@ export class ClinicalSourcePopup extends React.Component<PopupClinicalSourceProp
                                             className='margin-top-2'
                                             loading={this.state.addingOrEditingSource}
                                             onClick={this.addOrEdit}
-                                            disabled={isProcessing || !this.canAddOrEdit()}
+                                            disabled={isProcessing || !this.canAddOrEdit() || !this.validateSurvivalsColumnsComplete()}
```

Comment:
Si lo usás en dos lados entonces podés definirla como una constante del componente arriba de  `addOrEdit` como `const survColumnsAreComplete = this.validateSurvivalsColumnsComplete()` y usar esa constanste en todos lados, así solo se calcula una única vez por render

Reply:
Se utiliza en un solo lado del lado del componente, el otro es en la logica de un metodo, pero igual lo deje como const del lado del componente y de paso negarlo, asi solo usa el valor.